src/BotPlutusInterface/Effects.hs

```diff
@@ -152,6 +159,23 @@ printLog' :: LogLevel -> LogLevel -> String -> IO ()
 printLog' logLevelSetting msgLogLvl msg =
   when (logLevelSetting >= msgLogLvl) $ putStrLn msg
 
+-- | Version of "Control.Monad.Freer.Extras.handleLogTrace" that takes into account the log level setting.
+handleLogTrace' :: Pretty a => LogLevel -> Eff (LogMsg a ': effs) ~> Eff effs
+handleLogTrace' logLevelSetting = interpret $ \case
+  LMessage msg ->
+    if logLevelSetting >= toNativeLogLevel (msg ^. Freer.logLevel)
+      then Trace.trace (Render.renderString . layoutPretty defaultLayoutOptions . pretty $ msg) $ pure ()
```

Comment:
Logging with trace seems hacky - especially since PABEffect already has a logging effect, could we instead of handling this effect, reinterpret it in some way to PABEffect and use that code (which does direct `putStrLn`s in IO)
Alternatively, we can add the `LastMember` constraint on this, and lift to that.

Reply:
I agree. I just got lazy and decided to use the IOG `handleLogTrace`. I changed this to reinterpret the effect to use `PABEffect` later down the line.